src/hotspot/share/prims/jvmtiThreadState.inline.hpp

```diff
@@ -78,17 +78,27 @@ inline JvmtiThreadState* JvmtiThreadState::state_for_while_locked(JavaThread *th
 
   NoSafepointVerifier nsv;  // oop is safe to use.
 
-  if (thread_oop == nullptr) {  // Then thread should not be null (see assert above).
-    thread_oop = thread->jvmti_vthread() != nullptr ? thread->jvmti_vthread() : thread->threadObj();
-  }
-
   // In a case of unmounted virtual thread the thread can be null.
   JvmtiThreadState *state = thread == nullptr ? nullptr : thread->jvmti_thread_state();
 
   if (state == nullptr && thread != nullptr && thread->is_exiting()) {
     // Don't add a JvmtiThreadState to a thread that is exiting.
     return nullptr;
   }
```

Comment:
I'm wondering if there should also be an `is_jni_attaching` check here?

Reply:
That seems to be a good idea. It would cover other cases that we haven't seen yet. Added a check as suggested, thanks.